### PR TITLE
Add HAProxy PROXY header if configured for upstream handlers

### LINKED ISSUE #13: Supports HAProxy PROXY protocol in handler
It allows to keep the source address over any kind of protocol:

https://www.haproxy.com/blog/haproxy/proxy-protocol/

It is supported for example by Exim:

https://exim.org/exim-html-4.93/doc/html/spec_html/ch-proxies.html

### PR DESCRIPTION
**Pull request by mildred**
Follows #14
Fixes #13

### CODE REVIEW

**Comment by mildred** [comment]
Just checked it again, working great.

File: modules/l4proxy/proxy.go [hunk 1]
```diff
@@ -47,6 +48,9 @@ type Handler struct {
 	// Load balancing distributes load/connections between backends.
 	LoadBalancing *LoadBalancing `json:"load_balancing,omitempty"`
 
+	// Add HAPRoxy Proxy protocol header
+	ProxyHeader string `json:"proxy_header,omitempty"`
```
[the remark this answers is not in the document]
Comment (mholt) [reply modules/l4proxy/proxy.go:52]: I actually wonder if this should be an `int` instead, just the version number. Hmm.

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks for bringing it back! I think this is pretty good, just a couple nits. What are your thoughts on changing the field type from string to int, since it's just a version number? (I don't feel strongly either way. Why did you choose a string?)

File: modules/l4proxy/proxy.go [hunk 2]
```diff
@@ -73,6 +77,10 @@ func (h *Handler) Provision(ctx caddy.Context) error {
 		h.LoadBalancing.SelectionPolicy = mod.(Selector)
 	}
 
+	if h.ProxyHeader != "" && h.ProxyHeader != "v1" && h.ProxyHeader != "v2" {
+		return fmt.Errorf("proxy_header: \"%s\" should be empty or one of \"v1\" \"v2\"", h.ProxyHeader)
```
[the remark this answers is not in the document]
Comment (lxhao61) [reply modules/l4proxy/proxy.go:81]: totally agree！

File: modules/l4proxy/proxy.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mildred) [reply modules/l4proxy/proxy.go:52]: No specific reason why the string except empty value is more discoverable, and string is more future-proof.

**Review by mholt — APPROVED** [review]
Cool, thank you! Let's give it a try.

**Pull request merged**